docs/quickstarts/nextjs.mdx

```diff
@@ -115,31 +116,31 @@ Now that Clerk is installed and mounted in your application, you can decide whic
 
 1. Create a `middleware.ts` file at the root of your project, or in your `src/` directory if you have one.
 2. In your `middleware.ts`, export Clerk's [`clerkMiddleware`](/docs/references/nextjs/clerk-middleware) helper:
-    ```tsx filename="middleware.ts"
-    import { clerkMiddleware } from "@clerk/nextjs/server";
+  ```tsx filename="middleware.ts"
+  import { clerkMiddleware } from "@clerk/nextjs/server";
 
-    export default clerkMiddleware({});
-    ```
+  export default clerkMiddleware({});
+  ```
 3. With `clerkMiddleware()`, authentication will be enabled by default on all routes that your Next.js middleware runs on, blocking access to all signed out visitors. You can specify valid routes using Next.js's [`matcher`](https://nextjs.org/docs/app/building-your-application/routing/middleware#matcher). Add the following code to your `middleware.ts` to protect your entire application:
 
-    ```tsx filename="middleware.ts" {15}
-    import { clerkMiddleware } from "@clerk/nextjs/server";
-
-    export default clerkMiddleware({
-      // Routes that can be accessed while signed out
-      publicRoutes: ['/anyone-can-visit-this-route'],
-      // Routes that can always be accessed, and have
-      // no authentication information
-      ignoredRoutes: ['/no-auth-in-this-route'],
-    });
-
-    export const config = {
-      // Protects all routes, including api/trpc.
-      // See https://clerk.com/docs/references/nextjs/clerk-middleware
-      // for more information about configuring your Middleware
-      matcher: ["/((?!.+\\.[\\w]+$|_next).*)", "/", "/(api|trpc)(.*)"],
-    };
-    ```
+  ```tsx filename="middleware.ts" {15}
+  import { clerkMiddleware } from "@clerk/nextjs/server";
+
+  export default clerkMiddleware({
+    // Routes that can be accessed while signed out
+    publicRoutes: ['/anyone-can-visit-this-route'],
+    // Routes that can always be accessed, and have
+    // no authentication information
+    ignoredRoutes: ['/no-auth-in-this-route'],
+  });
+
+  export const config = {
+    // Protects all routes, including api/trpc.
+    // See https://clerk.com/docs/references/nextjs/clerk-middleware
+    // for more information about configuring your Middleware
+    matcher: ["/((?!.+\\.[\\w]+$|_next).*)", "/", "/(api|trpc)(.*)"],
+  };
+  ```

```

Comment:
formatting stuff again :/

Reply:
Yeah, I called this out in the main v5 PR where I think these changes were made directly. https://github.com/clerk/clerk-docs/pull/605#issuecomment-1953292508

And no, we 100% do not want to ship this. This is wrong code and wrong info. This could be merged into the v5 branch though and fixed before the v5 docs are shipped.

Reply:
maybe we can have this section say something like:

1. Create a `middleware.ts` file at the root of your project, or in your `src/` directory if you have one.
2. In your `middleware.ts`, export Clerk's [`clerkMiddleware()`](/docs/references/nextjs/clerk-middleware) helper:
  ```tsx filename="middleware.ts"
  import { clerkMiddleware } from "@clerk/nextjs/server";

  export default clerkMiddleware();

  export const config = {
    matcher: [
      '/((?!.*\\..*|_next).*)', // Don't run middleware on static files
      '/', // Run middleware on index page
      '/(api|trpc)(.*)'
    ], // Run middleware on API routes
  };
  ```
3. By default, `clerkMiddleware` will not protect any routes. All routes are public and you must opt-in to protection for routes. See the [`clerkMiddleware()`](/docs/references/nextjs/clerk-middleware) docs to learn how to require authentication for specific routes.

Reply:
My original version of this, which was merged into the now core2 branch, said something like this.

Reply:
oh I see it ! but I also think the code in core2 is incorrect? it reads:
```tsx filename="middleware.ts" {15}
    import { clerkMiddleware } from "@clerk/nextjs/server";

    export default clerkMiddleware({
      // Routes that can be accessed while signed out
      publicRoutes: ['/anyone-can-visit-this-route'],
      // Routes that can always be accessed, and have
      // no authentication information
      ignoredRoutes: ['/no-auth-in-this-route'],
    });

    export const config = {
      // Protects all routes, including api/trpc.
      // See https://clerk.com/docs/references/nextjs/clerk-middleware
      // for more information about configuring your Middleware
      matcher: ["/((?!.+\\.[\\w]+$|_next).*)", "/", "/(api|trpc)(.*)"],
    };
```

but does clerkMiddleware use publicRoutes and ignoredRoutes ?